### ISSUE TITLE
Destroy non-empty bucket

### ISSUE THREAD

**Issue by saulshanabrook** [hashicorp/terraform#1977]
Right now if you try to destroy an S3 bucket with any items in it, terraform will fail:

```
Error applying plan:

1 error(s) occurred:

* 1 error(s) occurred:

* BucketNotEmpty: The bucket you tried to delete is not empty

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

This is because S3 requires you delete all items before deleting the bucket. It would be great if terraform could do that automatically.

**Comment by radeksimko** [comment]
If we treat S3 like RDS which may have some data that we may not want to loose, then there would have to be either an option to do a backup or at least some kind of "force" option making sure that user is absolutely sure about removing data.

If we support removing data in buckets, then this behaviour IMO shouldn't be default (i.e. _by default_ Terraform should still show error from AWS API saying that bucket is not empty).

**Comment by mitchellh** [comment]
I think a force option makes sense, we do this with launch configurations or ASGs, I always mix up each.

**Comment by jimconner** [comment]
+1 for the force option - that would be very handy for our use case too.

**Comment by catsby** [comment]
I'm not seeing a `force` option of any kind in the API. We'd need to get a list and send the keys in a multi object delete API call.

**Comment by mitchellh** [comment]
@catsby Sorry thats what I meant. Its a TF setting only that we use to just iterate the bucket and do a multi-delete. Yep. Or, that's what we should do.

**Comment by m-s-austin** [comment]
https://github.com/hashicorp/terraform/pull/2007

**Comment by catsby** [comment]
I just merged #2007 which adds a `force_destroy` option, thanks!
